### ISSUE TITLE
legion-20.03.0: serializing test fails on i686

### ISSUE THREAD

**Issue by junghans** [StanfordLegion/legion#802]
From https://koji.fedoraproject.org/koji/taskinfo?taskID=42940048:
```
      Start 26: serializing
26/41 Test #26: serializing ......................***Failed    0.00 sec
ERROR: vector<PODPacked>dynamic size = 16 (should be 20)
ERROR: vector<PODPacked2>dynamic size = 13 (should be 17)
ERROR: map<int,double>dynamic size = 40 (should be 48)
ERRORS FOUND
```

**Comment by streichler** [comment]
@junghans has anything changed on the fedora side for this?  I'm pretty sure we test this in 32-bit mode in our CI and that nothing has changed related to this test since 19.12.0.  The error messages suggest that the checking logic thinks that `sizeof(size_t) == 8` for some reason.

**Comment by junghans** [comment]
Everything passed with 19.12.0.

**Comment by streichler** [comment]
Hmm...  I misread the polarity of the errors...  Here's what we get in our CI:
```
OK: vector<PODPacked> dynamic size = 20
OK: vector<PODPacked> dynamic output matches
OK: vector<PODPacked2> dynamic size = 17
OK: vector<PODPacked2> dynamic output matches
OK: list<int> dynamic size = 20
OK: list<int> dynamic output matches
OK: map<int,double> dynamic size = 48
OK: map<int,double> dynamic output matches
```
With the fedora test getting smaller sizes, the suspicion instead is that `__alignof__(double)` is returning something smaller than 8.  Are there any options in the Fedora build that might do that?

**Comment by streichler** [comment]
@junghans can I get a Dockerfile to repro this?  It's possible this is a c++98-vs-c++11 thing.

**Comment by junghans** [comment]
If you click on the link above it has the build.log including all flags.

**Comment by streichler** [comment]
I still can't reproduce this locally.  In the absence of a Dockerfile, can you try with the following patch:
https://gitlab.com/StanfordLegion/legion/-/commit/d4ef80b7678c1d84d2d110acbf40fd842a4d9aab.diff

**Comment by junghans** [comment]
Running https://koji.fedoraproject.org/koji/taskinfo?taskID=42953200

**Comment by streichler** [comment]
@junghans so when I ask what's changed on the Fedora side, mentioning that you're using a version of GCC that's not even been released yet is the sort of thing I'm looking for...  :)

**Comment by junghans** [comment]
19.02.0 was compiled with gcc-10 as well.

**Comment by streichler** [comment]
Ok, please try with this patch instead:
https://gitlab.com/StanfordLegion/legion/-/commit/3ea76cc5bd3f8afdced55c2d1659f53457c21320.diff

**Comment by junghans** [comment]
Running: https://koji.fedoraproject.org/koji/taskinfo?taskID=42954567

**Comment by junghans** [comment]
i686 seems to pass now!

**Comment by streichler** [comment]
Cool.  So are the Fedora packages ok with this patch and the one for #803?

**Comment by junghans** [comment]
Yeah I will add them in, just make sure they end up in the next release as well!

**Comment by streichler** [comment]
Yep, both of those are from a branch that'll merge into master shortly.  I'll close both bugs when that happens.

**Comment by streichler** [comment]
Fix is in the master branch with commit 293ce95d.
